addons/account/models/account_move.py

```diff
@@ -252,13 +252,11 @@ def _get_default_invoice_incoterm(self):
     @api.onchange('invoice_date')
     def _onchange_invoice_date(self):
         if self.invoice_date:
-            self.invoice_date_due = self.date = self.invoice_date
+            if not self.invoice_date_due and not self.invoice_payment_term_id:
```

Comment:
need to make sure that the using a payment term never sets a due date (before the `post()`)

Reply:
it doens't
if you put a payment term, but you also set an due date before, it will change to the correct date at pos